src/main.c

```diff
@@ -258,6 +258,9 @@ int cwstart = 0;			/**< number characters after which
 int sending_call = 0;
 int early_started = 0;			/**< 1 if sending call started early,
 					   strlen(hiscall)>cwstart or 'space' */
+int escape_pressed = 0;	        	/**< 0. Escape not last key pressed.
```

Comment:
I prefer using `bool` for actually boolean values. Also `x = true` or  `if (x) {..`/`if (!x) {..` makes code a bit more readable.

Reply:
Agreed.  Originally I'd planned on the values 0-3 for counting all presses until all fields are cleared.  Then I decided that isn't necessary.

As stated, this PR is mostly for review and discussion.  I say cherry pick anything that's good and improve/discard the rest as needed.